crates/domain/src/repositories/user_repository.rs

```diff
@@ -0,0 +1,102 @@
+use std::sync::Arc;
+
+use axum::async_trait;
+use sqlx::{query, Pool, Postgres, Row};
+use crate::models::user::User;
+use serde::{Serialize, Deserialize};
+
+use bcrypt::verify;
+use chrono::{Duration, Utc};
+use hmac::{Hmac, Mac};
+use jwt::SignWithKey;
+use sha2::Sha256;
+use base64;
+
+pub type DynUserRepository = Arc<dyn UserRepository + Send + Sync>;
+
+#[async_trait]
+pub trait UserRepository {
+    async fn user_login(&self, id: String, password: &str) -> anyhow::Result<String>;
```

Comment:
This is not a proper method for the repository. The repository is supposed to store/load the domain object, not perform an action.